spec/meilisearch/client/keys_spec.rb

```diff
@@ -1,60 +1,143 @@
-    expect do
-      new_client.keys
-    end.to raise_meilisearch_api_error_with(403, 'invalid_api_key', 'auth')
-  end
+    it 'fails to get keys if private key used' do
+      new_client = MeiliSearch::Client.new(URL, private_key['key'])
+      expect do
+        new_client.keys
+      end.to raise_meilisearch_api_error_with(403, 'invalid_api_key', 'auth')
+    end
 
-  it 'fails to search if no key used' do
-    new_client = MeiliSearch::Client.new(URL)
-    expect do
-      new_client.index(random_uid).settings
-    end.to raise_meilisearch_api_error_with(401, 'missing_authorization_header', 'auth')
-  end
+    it 'fails to search if no key used' do
+      new_client = MeiliSearch::Client.new(URL)
+      expect do
+        new_client.index(random_uid).settings
+      end.to raise_meilisearch_api_error_with(401, 'missing_authorization_header', 'auth')
+    end
 
-  it 'succeeds to search when using public key' do
-    uid = random_uid
-    public_key = client.keys['public']
-    index = client.create_index(uid)
-    response = index.add_documents(title: 'Test')
-    index.wait_for_pending_update(response['updateId'])
+    it 'succeeds to search when using public key' do
+      uid = random_uid
+      index = client.index(uid)
+      index.add_documents!(title: 'Test')
 
-    new_client = MeiliSearch::Client.new(URL, public_key)
-    response = new_client.index(uid).search('test')
-    expect(response).to have_key('hits')
-  end
+      new_client = MeiliSearch::Client.new(URL, public_key['key'])
+      response = new_client.index(uid).search('test')
+      expect(response).to have_key('hits')
+    end
 
-  it 'succeeds to get settings when using private key' do
-    uid = random_uid
-    client.create_index(uid)
-    private_key = client.keys['private']
-    new_client = MeiliSearch::Client.new(URL, private_key)
-    response = new_client.index(uid).settings
-    expect(response).to have_key('rankingRules')
+    it 'succeeds to get settings when using private key' do
+      uid = random_uid
+      client.create_index!(uid)
+      new_client = MeiliSearch::Client.new(URL, private_key['key'])
+      response = new_client.index(uid).settings
+      expect(response).to have_key('rankingRules')
+    end
   end
 
-  it 'works with method aliases' do
-    expect(client.method(:keys) == client.method(:get_keys)).to be_truthy
+  context 'Test the key managements' do
+    it 'gets the list of the default keys' do
+      results = client.keys['results']
+      expect(results).to be_a(Array)
+      expect(results.count).to be >= 2
+    end
+
+    it 'creates a key' do
+      key_options = {
+        description: 'A new key to add docs',
+        actions: ['documents.add'],
+        indexes: ['*'],
+        expiresAt: nil
+      }
+      new_key = client.create_key(key_options)
+      expect(new_key['expiresAt']).to be_nil
+      expect(new_key['key']).to be_a(String)
+      expect(new_key['createdAt']).to be_a(String)
```

Comment:
You mean only for the spec or the library in general?

Reply:
Well, I really like the idea to be a `Time`/`Date` (because of the idiomatic `_at` naming convention), so it would be awesome if in this PR we could make that parsing.

A discussion about that happened before https://github.com/meilisearch/meilisearch-ruby/pull/242. And I think some parts of the code already do that like https://github.com/meilisearch/meilisearch-ruby/blob/0e963ecec3d0b0e75ee3f4ca968912db8f51c6d4/spec/meilisearch/client/indexes_spec.rb#L169

In this case, it's your call if you have the time go ahead and do this conversion, if not, is ok to open an issue and we could refactor when we have the time 🦖